### ISSUE TITLE
It's possible to assist disabled disguised merchants

### ISSUE THREAD

**Issue by jostephd** [endless-sky/endless-sky#6466]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

It's possible to assist disguised merchants after disabling them.

### Steps to Reproduce

1. Take a "Disguised bandit" mission
2. Disable the bandit
3. Land
4. Take off
5. Board the bandit
6. The bandit becomes un-disabled

### Expected Behavior

6. Open the plundering dialog

### Screenshots

N/A

### Link to save file

[test3 test3.txt](https://github.com/endless-sky/endless-sky/files/7832480/test3.test3.txt)

Take off, jump one system (the flight plan is already set), board the disabled Manta.

### Operating System

N/A

### Game Source

Built from source

### Game Version

d05404714d6a6a258f3e2f3a6785baa6a8f4f188

### Additional Information

I haven't checked whether simply jumping out of the system and back (without an interim landing) would also cause the same behavior.

After repairing the bandit it's possible to disable it again and then plunder it as usual.

**Comment by Terin** [comment]
Yeah, this is a consequence of them holding no lasting grudges against you for attacking. We'd either have to have this, or have accepting a new mission here reset your rep with that faction, but if you picked up multiple missions, they'd all be hostile until you got a new one. We could make it so you can only get one Disguised Bandit mission at a time, but that'd be a bit rough for people who like to pick up loads of bounties at once.

**Comment by ThrawnCA** [comment]
For extra funny points, if you repeatedly repair and re-disable them, you can farm reputation with the Merchant faction. Maybe the Merchants enjoy having you torment the pirates?
